### PR TITLE
Rename getSslSocketFactoryForCertainCert to be newSslSocketFactoryForCa.

### LINKED ISSUE #560: Rename getSslSocketFactoryForCertainCert to be more clear what cert is provided
It isn't clear that the File should point to a CA certificate. It should be renamed to be clear.

`createSslSocketFactoryWithCa` or `newSslSocketFactoryForCa` or similar.

### PR DESCRIPTION
**Pull request by madongfly**
Fixes #560

### CODE REVIEW

**Comment by nmittler** [comment]
Why are we bothering to deprecate a method name in a testing utility?  This seems like overkill.

**Comment by ejona86** [comment]
@nmittler, because we know people are using it. Removing the API at the same time as adding its replacement is a major pain for upgrading. If we don't think the API is being used yet, then not messing with @Deprecated sounds fine, though.

File: testing/src/main/java/io/grpc/testing/TestUtils.java [hunk 1]
```diff
@@ -220,21 +220,29 @@ public static File loadCert(String name) throws IOException {
...
+   * Deprecated, please use {@link #newSslSocketFactoryForCa(File)} instead.
    */
+  @Deprecated
   public static SSLSocketFactory getSslSocketFactoryForCertainCert(File certChainFile)
           throws Exception {
+    return newSslSocketFactoryForCa(certChainFile);
+  }
+
+  /**
+   * Creates a SSLSocketFactory which uses the certificate specified in certChainFile.
```
Comment (ejona86) [inline testing/src/main/java/io/grpc/testing/TestUtils.java]: Please improve the comment. What does it "use the certificate" for?

Maybe:
```
Creates an SSLSocketFactory which contains {@code certChainFile} as its only root certificate.
```

Comment (madongfly) [inline testing/src/main/java/io/grpc/testing/TestUtils.java]: Done.

**Comment by madongfly** [comment]
@ejona86 PTAL, thanks!

**Comment by ejona86** [comment]
@madongfly LGTM

**Pull request merged**